framework/bootstrap/configstore.go

```diff
@@ -0,0 +1,350 @@
+package bootstrap
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/jsonpb"
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	"slime.io/slime/framework/bootstrap/resource"
+)
+
+var (
+	errNotFound      = errors.New("item not found")
+)
+
+const ResourceVersion string = "ResourceVersion"
+
+type ConfigStore interface {
+	// Schemas exposes the configuration type schema known by the config store.
+	Schemas() resource.Schemas
+	// Get retrieves a configuration element by a type and a key
+	Get(gvk resource.GroupVersionKind, name, namespace string) *resource.Config
+	// List returns objects by type and namespace.
+	// Use "" for the namespace to list across namespaces.
+	List(typ resource.GroupVersionKind, namespace string) ([]resource.Config, error)
+	// Create adds a new configuration object to the store. If an object with the
+	// same name and namespace for ProtoMessagethe type already exists, the operation fails
+	// with no side effects.
+	Create(config resource.Config) (revision string, err error)
+	// Update modifies an existing configuration object in the store.  Update
+	// requires that the object has been created.  Resource version prevents
+	// overriding a value that has been changed between prior _Get_ and _Put_
+	// operation to achieve optimistic concurrency. This method returns a new
+	// revision if the operation succeeds.
+	Update(config resource.Config) (newRevision string, err error)
+	// Delete removes an object from the store by key
+	Delete(typ resource.GroupVersionKind, name, namespace string) error
+}
+
+// NodeMetadata defines the metadata associated with a proxy
+// Fields should not be assumed to exist on the proxy, especially newly added fields which will not exist
+// on older versions.
+// The JSON field names should never change, as they are needed for backward compatibility with older proxies
+// nolint: maligned
+type nodeMetadata struct {
```

Comment:
`nodeMetadata` has nothing to do with `configstore`

Reply:
move to resource/context.go